cmake/version.cmake

```diff
@@ -0,0 +1,37 @@
+# bgfx versioning scheme:
+# bgfx 1.104.7082
+#      ^ ^^^ ^^^^
+#      | |   +--- Commit number  (https://github.com/bkaradzic/bgfx / git rev-list --count HEAD)
+#      | +------- API version    (from https://github.com/bkaradzic/bgfx/blob/master/scripts/bgfx.idl#L4)
+#      +--------- Major revision (always 1)
+
+# BGFX_API_VERSION generated from https://github.com/bkaradzic/bgfx/blob/master/scripts/bgfx.idl#L4
+# bgfx/src/version.h:
+# BGFX_REV_NUMBER
+# BGFX_REV_SHA1
+
+find_package(Git QUIET)
+
+execute_process(COMMAND "${GIT_EXECUTABLE}" -C bgfx log --pretty=format:'%h' -n 1
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+                OUTPUT_VARIABLE GIT_REV
+                ERROR_QUIET)
+
+execute_process(COMMAND "${GIT_EXECUTABLE}" -C bgfx rev-list --count HEAD
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+                OUTPUT_VARIABLE GIT_REV_COUNT
+                ERROR_QUIET)
+
+# read version(100) from bgfx.idl
+file(READ "${CMAKE_SOURCE_DIR}/bgfx/scripts/bgfx.idl" BGFX_IDL)
```

Comment:
This breaks when using as a fetch content (add external project) for a out of source build

```
Make Error at cmake-build-debug/_deps/bgfx-src/cmake/version.cmake:26 (file):
  file failed to open for reading (No such file or directory):

    ~/openblack/bgfx/scripts/bgfx.idl
```